src/auth/auth.service.ts

```diff
@@ -1,59 +1,35 @@
 import {
-  BadRequestException,
   ForbiddenException,
+  HttpException,
   Injectable,
   NotFoundException,
   UnauthorizedException,
 } from '@nestjs/common';
-import { JwtService, JwtSignOptions } from '@nestjs/jwt';
-import { BCRYPT, NUMBER_OF_2FA_RECOVERY_TOKENS, SERVICE } from './constants';
-import * as bcrypt from 'bcryptjs';
-import {
-  AccessTokenPayload,
-  SignInRequest,
-  SignUpRequest,
-  SignUpResponse,
-  UserRequest,
-} from './dto';
 import { UserRepository } from '../user/user.repository';
-import { UserPayloadRequest } from '../user/dto';
-import { PersonalAccessTokenRepository } from './personal-access-token.repository';
-import { authenticator } from 'otplib';
-import qrcode from 'qrcode';
-import { TwoFactorAuth } from '@prisma/client';
-import { TwoFactorAuthRepository } from './twoFactorAuth.repository';
 import { PendingUsersRepository } from '../user/pending-user.repository';
+import { UserPayloadRequest } from '../user/dto';
+import { SignInRequest, SignUpRequest, UserRequest } from './dto';
+import { AmqpConnection } from '@golevelup/nestjs-rabbitmq';
 
 @Injectable()
 export class AuthService {
   constructor(
     private readonly pendingUsersRepository: PendingUsersRepository,
     private readonly userRepository: UserRepository,
-    private readonly twoFactorAuthRepository: TwoFactorAuthRepository,
-    private readonly personalAccessTokenRepository: PersonalAccessTokenRepository,
-    private readonly jwtService: JwtService,
+    private readonly amqpConnection: AmqpConnection,
   ) {}
 
-  verifyJwt(jwtToken: string, secret: string): Promise<AccessTokenPayload> {
-    return this.jwtService.verifyAsync(jwtToken, { secret });
-  }
-
-  async generateToken(
-    id: number,
-    secret: string,
-    time?: string,
-  ): Promise<string> {
-    const payload = { id };
-    const options: JwtSignOptions = { secret };
+  private EXCHANGE = 'authentication';
 
-    if (time) {
-      options.expiresIn = time;
+  validateAuthMicroserviceReturn(payload: any) {
```

Comment:
`any` warning - if payload can really be anything, it's better to use `unknown`. However I see you are expecting `message` and `status` fields inside, so there must be some certain type passed here. If there can be multiple types passed here then you can use unions, e.g. `TypeA | TypeB`. If inside the type there are few fields defined and there can be other unknown fields, you can use something like this: `[key: string]: unknown` (example [here](https://johndpelingo.com/notes/typescript-known-and-unknown-keys/))